pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentDispatcherMultipleConsumers.java

```diff
@@ -191,7 +191,7 @@ public RedeliveryTracker getRedeliveryTracker() {
     }
 
     @Override
-    public void sendMessages(List<Entry> entries) {
+    public synchronized void sendMessages(List<Entry> entries) {
```

Comment:
I think let  `getNextConsumer()` synchronized  is enough:)

Reply:
we have about 4 places called `getNextConsumer()`,  if we add `synchronized ` in the `getNextConsumer ()`, we don't know if there are performance issue.